Sources/IRGen/Preprocessor/IRPreprocessor.swift

```diff
@@ -29,6 +29,15 @@ public struct IRPreprocessor: ASTPass {
     return ASTPassResult(element: structMember, diagnostics: [], passContext: passContext)
   }
 
+  public func process(traitDeclaration: TraitDeclaration,
```

Comment:
Perhaps move this into the previous pass (`IRConformanceProcessor` or whatever its final name will be), so that traits are completely "resolved" after that pass and the IR doesn't even see them?